src/javascript/request-composer/request.js

```diff
@@ -0,0 +1,318 @@
+// Copyright 2016 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import moment from 'moment';
+
+function buildReportRequest(params) {
+  let reportRequest = {
+      viewId: params.viewId,
+      samplingLevel: params.samplingLevel,
+  };
+  if (params.filters) {
+    reportRequest.filtersExpression = params.filters;
+  }
+  if (params.includeEmptyRows) {
+    if (params.includeEmptyRows.toLowerCase() == 'true') {
+      reportRequest.includeEmptyRows = 'true';
+    } else if (params.includeEmptyRows.toLowerCase() == 'false') {
+      reportRequest.includeEmptyRows = 'false';
+    }
+  }
+  if (params.pageSize &&
+      parseInt(params.pageSize)) {
+    reportRequest.pageSize = params.pageSize;
+  }
+  if (params.pageToken) {
+    reportRequest.pageToken = params.pageToken;
+  }
+  return reportRequest;
+}
+
+function applyDateRanges(request, params, settings) {
+  if (params.startDate &&
+      params.endDate &&
+      settings.requestType != 'COHORT') {
+    request.dateRanges = [{
+      'startDate': params.startDate,
+      'endDate': params.endDate
+    }];
+  }
+  return request;
+}
+
+function applyMetrics(request, params, settings) {
+  if (params.metrics &&
+      settings.requestType != 'COHORT') {
+    request.metrics = [];
+
+    let metrics = params.metrics.split(',');
+    for (let i in metrics) {
```

Comment:
You should use [for...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) loops here (and everywhere else).

```js
for (let metric of metrics) {
  request.metrics.push({expression: metric});
}
```

For..in loops return the key instead of the value and have issues where they also iterate over properties up the prototype chain.